### ISSUE TITLE
Add code of conduct information to CSV

### ISSUE THREAD

**Issue by chrisjrn** [python-organizers/conferences#102]
The current list does not include code of conduct information. Codes of Conduct are a sign that the conference takes the safety of attendees seriously, and are an expected part of Python conferences. The PSF does not offer grants to events that do not have a functional code of conduct.

It is also problematic to promote conferences whose enforcement processes are deficient.

This list should:
* require a link to the CoC for the current conference to the CoC
* remove conferences that do not have public CoC information
* provide an avenue to confidentially report conferences who do not take their CoC and enforcement seriously, so that they can be removed from the list without revealing the reporter's identity

**Comment by Mariatta** [comment]
Thanks @chrisjrn. I agree that we should add coc link to each conferences.
I'll welcome PR is welcome (by anyone).

> provide an avenue to confidentially report conferences who do not take their CoC and enforcement seriously, so that they can be removed from the list without revealing the reporter's identity

I'm wondering what's the best way to do this. Write email to me to raise the issue? (if people want to trust me on this) or write to The PSF? 🤔

**Comment by chrisjrn** [comment]
I don't think repository is an official PSF resource (is it?), but I can see if the PSF code of conduct working group has the capacity to look over this resource?

**Comment by jefftriplett** [comment]
I would be in favor of adding a column to note which websites have a CoC. Since this is not an official PSF repo/org, the goal was to collect info on the regional conferences at large. 

If the PSF/PyCon wants to consume this data to display on the website (which I think was Ernest's goal and I hope that they/we will) then I would recommend that we only show websites which a) have a CoC and b) publicly address @chrisjrn's concerns about safety *on the PSF/PyCon website*.

**Comment by chrisjrn** [comment]
@jefftriplett Whilst I am in favour of adding a link to a confernece's CoC, note that without a proven track record of enforcement, a CoC can be a false indicator of attendee safety. This list should not display the code of conduct for an event that does not take its code of conduct seriously, and there should be a process for reporting such events.

**Comment by jonafato** [comment]
>require a link to the CoC for the current conference to the CoC

:+1: I think this is an easy first step. The first PR should probably be a CSV file format change to include a new column and a note that it's required in the readme. The only place this gets tricky is conferences that don't yet have websites, e.g. entries frequently end up on this list because dates have been announced but no other information. In those cases, I'd propose accepting a prior year's code of conduct link or allowing it to be empty in the case of a new conference (e.g. if PyMetropolis announces dates for their inaugural conference without a website, it should probably be allowed in the list until the website launches, barring other historical bad behavior).

> remove conferences that do not have public CoC information

Should open an issue and @-mention everyone who's added a conference to the list here? Other thoughts on how to get this done quickly and easily? A PR that removes conferences before giving them a chance to add their links wouldn't be very helpful. 

> > provide an avenue to confidentially report conferences who do not take their CoC and enforcement seriously, so that they can be removed from the list without revealing the reporter's identity

> I'm wondering what's the best way to do this. Write email to me to raise the issue? (if people want to trust me on this) or write to The PSF? :thinking: 

I _really_ hesitate to put the burden of receiving code of conduct reports for arbitrarily many conferences on a single person, which would be a fast lane to burnout. I don't know what the correct answer is here, but I'm sure it requires a group with well-formed policies. If the PSF wants to get involved, I think that's a discussion to have, but it shouldn't prevent us from implementing the first steps.

**Comment by invisibleroads** [comment]
Please do not remove Python conferences from this repository.
The purpose of this repository is to make it easier for people to find
Python conferences around the world.

Having a code of conduct is an important requirement for the PSF to enforce
as a condition for receiving PSF funding for an event.

However, I do not think we should add another column to this dataset unless
absolutely necessary.

**Comment by Mariatta** [comment]
I agree that having enforceable CoC is important but at the same time, it can be disadvantageous to new conference.

Also, I'm curious to know how do we measure/judge whether a conference is taking this seriously? Is it by personal experience? Gossip?  Do we have a checklist, like - they have a hotline, they published CoC transparency report, they do this and that?

Do we want to exclude communities this way (by not listing them), or we just add a big notice in readme, like * warning this conference is shady because of CoC*, and leave it up to the attendee to decide?

**Comment by tylerdave** [comment]
I agree with listing CoC as an optional field for the purpose of this repo. There was discussion of using this repo (at least initially) as a source for the pycon.org listing and I think that's where this becomes an issue. By listing events on the official website the PSF is implying endorsement. I think the PSF needs to decide on a solution for vetting/approving conferences to be on the list on that site and that's separate from the goals of this repo.
